### PR TITLE
fix: improve provisioner key cli usability

### PR DESCRIPTION
**Pull request by f0ssel**
What this changes:
- Unhides the `--key` flag on provisioner start
- Deprecates and hides `provisionerd` command group in favor of `provisioner(s)`
- Removes org id from `coder provisioner keys list`

### CODE REVIEW

File: enterprise/cli/provisionerkeys.go [hunk 1]
```diff
@@ -144,6 +145,29 @@ func (r *RootCmd) provisionerKeysList() *serpent.Command {
...
+		}
+		out = append(out, provisionerKeyListItem{
+			Name:      item.Name,
+			CreatedAt: item.CreatedAt,
+			Tags:      strings.Join(tags, ", "),
+		})
+	}
+
+	return out
+}
```
Comment (Emyrk) [inline enterprise/cli/provisionerkeys.go:169]: We should use the original keys, just change the default columns.

```golang
cliui.TableFormat([]codersdk.ProvisionerKey{}, []string{"Created At", "Name", "Tags"}),
```

Then you can optionally grab the other columns if you'd like with `-c`

File: enterprise/cli/provisionerkeys.go [hunk 2]
```diff
@@ -95,7 +96,7 @@ func (r *RootCmd) provisionerKeysList() *serpent.Command {
 	var (
 		orgContext = agpl.NewOrganizationContext()
 		formatter  = cliui.NewOutputFormatter(
-			cliui.TableFormat([]codersdk.ProvisionerKey{}, nil),
+			cliui.TableFormat([]provisionerKeyListItem{}, nil),
```
Comment (Emyrk) [inline enterprise/cli/provisionerkeys.go:99]:
```suggestion
			cliui.TableFormat([]codersdk.ProvisionerKey{}, []string{"created at", "name", "tags"}),
```

File: enterprise/cli/provisionerkeys.go (hunk 1 above)
Comment (f0ssel) [reply enterprise/cli/provisionerkeys.go:169]: I need to convert tags render from `map[my:val,your:val]` to `my=val, your=val` which is why I made the new struct.

Comment (Emyrk) [reply enterprise/cli/provisionerkeys.go:169]: Ah. IIRC we call `.String()` if present on the type.

We could make a type, and implement `String()` on it.

```golang
type ProvisionerKeyTags map[string]string

func (p ProvisionerKeyTags) String() string {
	return "yay"
}

type ProvisionerKey struct {
	ID             uuid.UUID         `json:"id" table:"-" format:"uuid"`
	CreatedAt      time.Time         `json:"created_at" table:"created at" format:"date-time"`
	OrganizationID uuid.UUID         `json:"organization" table:"organization id" format:"uuid"`
	Name           string            `json:"name" table:"name,default_sort"`
	Tags           ProvisionerKeyTags `json:"tags" table:"tags"`
	// HashedSecret - never include the access token in the API response
}
```

I prefer not to make type duplicates for formatting where we can avoid it.

**Review by Emyrk — APPROVED** [review]

**Pull request merged**